### ISSUE TITLE
OSError: broken data stream when reading image file

### ISSUE THREAD

**Issue by marybethcassity** [DeepLabCut/DeepLabCut#2531]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Recieving common PILLOW error when training.  See this forum [https://github.com/python-pillow/Pillow/issues/1510](https://github.com/python-pillow/Pillow/issues/1510). I am using PILLOW 10.2.0. I tried to update PILLOW, but received that PILLOW was up to date. If not a common DLC issue or due to PILLOW version, must be corrupted image file.

### Operating System

Microsoft Windows 10 Enterprise N

### DeepLabCut version

2.3.8

### DeepLabCut mode

single animal

### Device type

NVIDIA RTX A6000

### Steps To Reproduce

Use Train Network in DeepLabCut gui. 

### Relevant log output

```shell
iteration: 901000 loss: 0.0015 lr: 0.02
Exception in thread Thread-5:
Traceback (most recent call last):
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\threading.py", line 980, in _bootstrap_inner
    self.run()
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\core\train.py", line 84, in load_and_enqueue
    batch_np = dataset.next_batch()
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\datasets\pose_imgaug.py", line 418, in next_batch
    ) = self.get_batch()
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\datasets\pose_imgaug.py", line 347, in get_batch
    raise e
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\datasets\pose_imgaug.py", line 343, in get_batch
    image = imread(image_path, mode="skimage")
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\utils\auxfun_videos.py", line 371, in imread
    image = io.imread(image_path)
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\skimage\io\_io.py", line 53, in imread
    img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\skimage\io\manage_plugins.py", line 205, in call_plugin
    return func(*args, **kwargs)
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\skimage\io\_plugins\imageio_plugin.py", line 11, in imread
    out = np.asarray(imageio_imread(*args, **kwargs))
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\imageio\v3.py", line 54, in imread
    return np.asarray(img_file.read(**call_kwargs))
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\imageio\plugins\pillow.py", line 252, in read
    image = self._apply_transforms(
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\imageio\plugins\pillow.py", line 333, in _apply_transforms
    image = np.asarray(image)
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\PIL\Image.py", line 681, in __array_interface__
    new["data"] = self.tobytes()
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\PIL\Image.py", line 740, in tobytes
    self.load()
  File "C:\Users\Admin\anaconda3\envs\deeplabcut\lib\site-packages\PIL\ImageFile.py", line 310, in load
    raise _get_oserror(err_code, encoder=False)
OSError: broken data stream when reading image file
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
Hi @marybethcassity! I think the most likely issue is that one of the images is corrupted. Have you been able to check that you can open all images normally?

And could you check if running `Check labels` (with the button in the "Label Frames" tab of the GUI works correctly)?

**Comment by marybethcassity** [comment]
@n-poulsen  Check labels works fine! It was a corrupted image-- I edited pose_imgaug.py to check if the image being read is corrupted. If it is, it prints the image path and replaces it with another random image. Weirdly, on my next project iteration the image was fine and not giving me any trouble. I'm not sure why-- when opening it in Jupyter Notebook/ File Explorer it still looks corrupted, but can be read. 

Edit: Would it be best to replace with another random image, or the frame before or after the corrupted image? I want the model to see about the same amount of variation.

**Comment by n-poulsen** [comment]
@marybethcassity I think you should just remove that image from the dataset completly (or if you can find a way to extract it without it being corrupted that would be even better). It will really confuse the model if your keypoints are labelled normally, but the content of the image is corrupted (if the image looks good in file explorer then all good!).

You should be able to remove the corrupt image from the dataset by:

- deleting the corrupt image from the `labeled-data` folder
- open the napari labelling tool
- re-saving the data

**Comment by marybethcassity** [comment]
@n-poulsen I should specify that it replaces it with another random image + that random image's keypoints. If I did it correctly, the model shouldn't see a corrupt image or mismatched keypoints. Will remove the image from the dataset per your instructions! Thanks! But also, what I implemented is to detect if images are corrupted and tell me which ones are-- I didn't know this one was until I started training and had the error several times.
